From 5b58141396d03305e9db380648800da19ac5d960 Mon Sep 17 00:00:00 2001
From: Denis V. Lunev <den@openvz.org>
Date: Mon, 14 Jul 2008 11:04:29 +0400
Subject: [PATCH 81/82] ubc: uncharging too much for TCPSNDBUF

It is not allowed to go to the label wait_for_memory with chargesize != 0
when this space is already placed to the skb.

Signed-off-by: Denis V. Lunev <den@openvz.org>
Signed-off-by: Pavel Emelyanov <xemul@openvz.org>
---
 net/ipv4/tcp.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/net/ipv4/tcp.c b/net/ipv4/tcp.c
index 56f3de7..a68ebd3 100644
--- a/net/ipv4/tcp.c
+++ b/net/ipv4/tcp.c
@@ -723,6 +723,7 @@ new_segment:
 			if (!skb)
 				goto wait_for_memory;
 			ub_skb_set_charge(skb, sk, chargesize, UB_TCPSNDBUF);
+			chargesize = 0;
 
 			skb_entail(sk, skb);
 			copy = size_goal;
@@ -915,6 +916,7 @@ new_segment:
 					goto wait_for_memory;
 				ub_skb_set_charge(skb, sk, chargesize,
 						UB_TCPSNDBUF);
+				chargesize = 0;
 
 				/*
 				 * Check whether we can use HW checksum.
-- 
1.6.2.2

